x-pack/platform/plugins/shared/aiops/public/components/change_point_detection/use_change_point_agg_request.ts

```diff
@@ -223,43 +230,81 @@ export function useChangePointResults(
-              label: changePointType,
-              reason: v.change_point_request.type[changePointType].reason,
-              id: isSingleMetric
-                ? 'single_metric'
-                : `${fieldConfig.splitField}_${v.key?.splitFieldTerm}`,
-            } as ChangePointAnnotation;
-          })
-          .filter((v) => !EXCLUDED_CHANGE_POINT_TYPES.has(v.type));
+        const currentRawChangePoints = buckets.map((v) => {
+          const changePointType = Object.keys(v.change_point_request.type)[0] as ChangePointType;
+          const timeAsString = v.change_point_request.bucket?.key;
+          const rawPValue = v.change_point_request.type[changePointType].p_value;
+
+          return {
+            ...(isSingleMetric
+              ? {}
+              : {
+                  group: {
+                    name: fieldConfig.splitField,
+                    value: v.key.splitFieldTerm,
+                  },
+                }),
+            kind: 'changePoint',
+            type: changePointType,
+            p_value: rawPValue,
+            timestamp: timeAsString,
+            label: changePointType,
+            reason: v.change_point_request.type[changePointType].reason,
+            id: isSingleMetric
+              ? 'single_metric'
+              : `${fieldConfig.splitField}_${v.key?.splitFieldTerm}`,
+          } as ChangePointAnnotation;
+        });
+
+        // Store first raw change point from first request
+        if (pageNumber === 1 && !firstRawChangePoint.current && currentRawChangePoints.length > 0) {
+          firstRawChangePoint.current = currentRawChangePoints[0];
+        }
+
+        // Filter for real change points
+        let currentValidChangePoints = currentRawChangePoints.filter(
+          (v) => v.kind === 'changePoint' && !EXCLUDED_CHANGE_POINT_TYPES.has(v.type)
+        );
 
         if (Array.isArray(requestParams.changePointType)) {
-          groups = groups.filter((v) => requestParams.changePointType!.includes(v.type));
+          currentValidChangePoints = currentValidChangePoints.filter(
+            (v) => v.kind === 'changePoint' && requestParams.changePointType!.includes(v.type)
+          );
         }
 
-        setResults((prev) => {
-          return (prev ?? []).concat(groups);
+        setValidChangePoints((prev) => {
+          const hasNoValidChangePoints = prev.length === 0 && currentValidChangePoints.length === 0;
+
+          // If there are no valid change points, and we have data, and the fetches are complete,
+          // return the first raw change point as no change points found
+          if (hasNoValidChangePoints && isFetchCompleted && firstRawChangePoint.current) {
+            return [
+              {
+                ...firstRawChangePoint.current,
+                kind: 'noChangePoints',
+                label: 'No change points found',
+                p_value: null,
+                timestamp: null,
+              },
+            ];
```

Comment:
Can we simply return an empty array here instead? And get rid of the additional type with `kind: 'noChangePoints` ? 
It should simplify the code quite a bit.